File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        i=0
        j=0
        l=len(s)
        max_freq=float('-inf')        
        max_len=float('-inf')
        freq={}
        while(j<l):
            freq[s[j]]=freq.get(s[j],0)+1
            max_freq=max(max_freq,freq[s[j]])
            while((j-i+1)-max_freq>k):
                freq[s[i]]-=1
                i+=1
            max_len=max(max_len,j-i+1)
            j+=1
        return max_len

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int i=0;
        int j=0;
        int len=s.size();
        int max_freq=INT_MIN;
        int max_len=INT_MIN;
        vector<int> freq(256,0);
        while(j<len)
        {
            freq[s[j]]++;
            max_freq=max(max_freq,freq[s[j]]);
            while((j-i+1)-max_freq>k)
            {
                freq[s[i]]--;
                i++;
            }
            max_len=max(max_len,j-i+1);
            j++;
        }
        return max_len;
    }
};